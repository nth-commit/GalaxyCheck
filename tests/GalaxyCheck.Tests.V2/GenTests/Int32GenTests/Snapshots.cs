using System.Collections.Generic;
using System.Linq;
using Xunit;
using GalaxyCheck;
using GalaxyCheck.Gens;
using Snapshooter.Xunit;
using Snapshooter;

namespace Tests.V2.GenTests.Int32GenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
                { "Between_0_And_10000", Gen.Int32().Between(0, 10000) },
                { "Between_Default", Gen.Int32() },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen.WithBias(bias).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 5);

            var gens = new Dictionary<string, IInt32Gen>
            {
                { "Between_0_And_10", Gen.Int32().Between(0, 10) },
                { "Between_0_And_100", Gen.Int32().Between(0, 100) },
            };

            var biases = new[] { Gen.Bias.None, Gen.Bias.WithSize };

            foreach (var seed in seeds)
            foreach (var (betweenDescription, gen) in gens)
            foreach (var bias in biases)
            {
                var sample = gen
                    .WithBias(bias)
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x.ToString());

                var nameExtension = string.Join("_", new[]
                {
                    betweenDescription,
                    $"Bias_{bias}",
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
