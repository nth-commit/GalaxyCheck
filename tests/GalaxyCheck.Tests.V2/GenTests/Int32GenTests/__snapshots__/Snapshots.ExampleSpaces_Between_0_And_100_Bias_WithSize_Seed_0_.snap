28 (Id = 28, Distance = 28.00)
├> 0 (Id = 0, Distance = 0)
├> 14 (Id = 14, Distance = 14.00)
|  ├> 7 (Id = 7, Distance = 7.00)
|  |  ├> 4 (Id = 4, Distance = 4.00)
|  |  |  ├> 2 (Id = 2, Distance = 2.00)
|  |  |  |  └> 1 (Id = 1, Distance = 1.00)
|  |  |  └> 3 (Id = 3, Distance = 3.00)
|  |  └> 6 (Id = 6, Distance = 6.00)
|  |     ├> 3 (Id = 3, Distance = 3.00)
|  |     |  └> 2 (Id = 2, Distance = 2.00)
|  |     |     └> 1 (Id = 1, Distance = 1.00)
|  |     └> 5 (Id = 5, Distance = 5.00)
|  ├> 11 (Id = 11, Distance = 11.00)
|  |  ├> 6 (Id = 6, Distance = 6.00)
|  |  |  ├> 3 (Id = 3, Distance = 3.00)
|  |  |  |  └> 2 (Id = 2, Distance = 2.00)
|  |  |  |     └> 1 (Id = 1, Distance = 1.00)
|  |  |  └> 5 (Id = 5, Distance = 5.00)
|  |  |     └> 4 (Id = 4, Distance = 4.00)
|  |  |        └> 2 (Id = 2, Distance = 2.00)
|  |  |           └> 1 (Id = 1, Distance = 1.00)
|  |  ├> 9 (Id = 9, Distance = 9.00)
|  |  |  ├> 5 (Id = 5, Distance = 5.00)
|  |  |  |  ├> 3 (Id = 3, Distance = 3.00)
|  |  |  |  |  └> 2 (Id = 2, Distance = 2.00)
|  |  |  |  |     └> 1 (Id = 1, Distance = 1.00)
|  |  |  |  └> 4 (Id = 4, Distance = 4.00)
|  |  |  |     └> 2 (Id = 2, Distance = 2.00)
|  |  |  |        └> 1 (Id = 1, Distance = 1.00)
|  |  |  └> 8 (Id = 8, Distance = 8.00)
|  |  |     └> 4 (Id = 4, Distance = 4.00)
|  |  |        ├> 2 (Id = 2, Distance = 2.00)
|  |  |        |  └> 1 (Id = 1, Distance = 1.00)
|  |  |        └> 3 (Id = 3, Distance = 3.00)
|  |  └> 10 (Id = 10, Distance = 10.0)
|  |     ├> 5 (Id = 5, Distance = 5.00)
|  |     |  ├> 3 (Id = 3, Distance = 3.00)
|  |     |  |  └> 2 (Id = 2, Distance = 2.00)
|  |     |  |     └> 1 (Id = 1, Distance = 1.00)
|  |     |  └> 4 (Id = 4, Distance = 4.00)
|  |     |     └> 2 (Id = 2, Distance = 2.00)
|  |     |        └> 1 (Id = 1, Distance = 1.00)
|  |     └> 8 (Id = 8, Distance = 8.00)
|  |        └> 4 (Id = 4, Distance = 4.00)
|  |           ├> 2 (Id = 2, Distance = 2.00)
|  |           |  └> 1 (Id = 1, Distance = 1.00)
|  |           └> 3 (Id = 3, Distance = 3.00)
|  └> 13 (Id = 13, Distance = 13.00)
|     ├> 10 (Id = 10, Distance = 10.0)
|     |  ├> 5 (Id = 5, Distance = 5.00)
|     |  |  ├> 3 (Id = 3, Distance = 3.00)
|     |  |  |  └> 2 (Id = 2, Distance = 2.00)
|     |  |  |     └> 1 (Id = 1, Distance = 1.00)
|     |  |  └> 4 (Id = 4, Distance = 4.00)
|     |  |     └> 2 (Id = 2, Distance = 2.00)
|     |  |        └> 1 (Id = 1, Distance = 1.00)
|     |  ├> 8 (Id = 8, Distance = 8.00)
|     |  |  ├> 4 (Id = 4, Distance = 4.00)
|     |  |  |  ├> 2 (Id = 2, Distance = 2.00)
|     |  |  |  |  └> 1 (Id = 1, Distance = 1.00)
|     |  |  |  └> 3 (Id = 3, Distance = 3.00)
|     |  |  └> 6 (Id = 6, Distance = 6.00)
|     |  |     └> 3 (Id = 3, Distance = 3.00)
|     |  |        └> 2 (Id = 2, Distance = 2.00)
|     |  |           └> 1 (Id = 1, Distance = 1.00)
|     |  └> 9 (Id = 9, Distance = 9.00)
|     └> 12 (Id = 12, Distance = 12.00)
|        ├> 6 (Id = 6, Distance = 6.00)
|        |  ├> 3 (Id = 3, Distance = 3.00)
|        |  |  └> 2 (Id = 2, Distance = 2.00)
|        |  |     └> 1 (Id = 1, Distance = 1.00)
|        |  └> 5 (Id = 5, Distance = 5.00)
|        |     └> 4 (Id = 4, Distance = 4.00)
|        |        └> 2 (Id = 2, Distance = 2.00)
|        |           └> 1 (Id = 1, Distance = 1.00)
|        └> 9 (Id = 9, Distance = 9.00)
|           ├> 5 (Id = 5, Distance = 5.00)
|           |  ├> 3 (Id = 3, Distance = 3.00)
|           |  |  └> 2 (Id = 2, Distance = 2.00)
|           |  |     └> 1 (Id = 1, Distance = 1.00)
|           |  └> 4 (Id = 4, Distance = 4.00)
|           |     └> 2 (Id = 2, Distance = 2.00)
|           |        └> 1 (Id = 1, Distance = 1.00)
|           └> 8 (Id = 8, Distance = 8.00)
|              └> 4 (Id = 4, Distance = 4.00)
|                 ├> 2 (Id = 2, Distance = 2.00)
|                 |  └> 1 (Id = 1, Distance = 1.00)
|                 └> 3 (Id = 3, Distance = 3.00)
├> 21 (Id = 21, Distance = 21.00)
|  ├> 11 (Id = 11, Distance = 11.00)
|  |  ├> 6 (Id = 6, Distance = 6.00)
|  |  |  ├> 3 (Id = 3, Distance = 3.00)
|  |  |  |  └> 2 (Id = 2, Distance = 2.00)
|  |  |  |     └> 1 (Id = 1, Distance = 1.00)
|  |  |  └> 5 (Id = 5, Distance = 5.00)
|  |  |     └> 4 (Id = 4, Distance = 4.00)
|  |  |        └> 2 (Id = 2, Distance = 2.00)
|  |  |           └> 1 (Id = 1, Distance = 1.00)
|  |  ├> 9 (Id = 9, Distance = 9.00)
